teuthology/task/internal/redhat.py

```diff
@@ -88,6 +88,36 @@ def setup_cdn_repo(ctx, config):
         set_cdn_repo(ctx, config)
     yield
 
+@contextlib.contextmanager
+def setup_container_registry(ctx, config):
+    """
+     setup container registry if setup_container_registry in config
+
+     redhat:
+      setup_container_registry: <registry.io>   # registry-name
+    """
+    if ctx.config.get('redhat').get('setup_container_registry', None):
+        registry = ctx.config['redhat']['setup_container_registry']
+
+        # fetch credentials from teuth_config
+        creds = teuthconfig.get('registries', dict()).get(registry)
+        if not creds:
+            raise ConfigError("Registry not found....")
+
+        # container-tool login
+        for remote in ctx.cluster.remotes.keys():
+            container_tool = "docker"
+            if remote.os.version.startswith('8'):
```

Comment:
instead of checking the version number, which would need updating each release, this could follow ceph-cm-ansible and use docker on deb-based and podman on rpm-based systems.

you can check for which type of system it is with teuthology.misc's get_system_type(remote) function